cmd/storage-rest-common.go

```diff
@@ -16,7 +16,7 @@
 
 package cmd
 
-const storageRESTVersion = "v9"
+const storageRESTVersion = "v10"
```

Comment:
As s2 compression is compatible with snappy compressed content. Why do we need to increment the version?

Reply:
To prevent a downgraded version to work. 

A v9 server will not be able to read v10 content, since S2 content cannot be decoded by Snappy, only the other way around. 

Does that make sense?

Reply:
Ah! yes. Understood :+1:

Reply:
This change is not needed, this prevents our users to not do rolling upgrade. But since we don't change any StorageAPI we don't need to do it. 

What we should do is have code to check for both types of algorithmV1 and algorithmV2 like  metadata validation. This would allow us to reject downgrades, meaning unexpected entry should be rejected. 

But changes in top level compression library doesn't mandate breakage in deployments, meaning without changing any Storage API we shouldn't update the version here.

Reply:
@harshavardhana I will ofc. remove it if it doesn't do what I want.

> What we should do is have code to check for both types of algorithmV1 and algorithmV2 like metadata validation. 

Yeah, but there is no way of checking this on older versions. I can add it for future versions, if at some point we want to add alternative schemes. Is that what you mean?

Reply:
That is perhaps the problem here.. we shall add validation. Also make the code to allow older metadata as well. Because both are valid and s2 can read both. Because if we do that we don't need to make storageAPI change here.

Reply:
And we can keep rolling upgrades..